### PR TITLE
feat: sync DApp provider with wallet provider

### PR DESCRIPTION
**Pull request by luizstacio**
closes: #393 #391

### CODE REVIEW

**Comment by LuizAsFight** [comment]
I'm testing in latest PR preview docs/extension

context: I added a new network "Localhost" pointing 

problems in connect screen:
- when click in "Localhost", nothing happens
- when click in "account" nothing happends
- after connecting, when trying to disconnect, an error is shown


https://user-images.githubusercontent.com/8636507/212160210-bd159fe6-cabf-4983-97bb-930a7918348c.mov

**Comment by LuizAsFight** [comment]
in same context previously mentioned:

when click in "transfer asset", a error shows up "Network request Failed".
Debugging it I see "ERR_BLOCKED_BY_CLIENT" error, probably configs between domains, some block from chrome because it's a localhost server... just wanna make sure this is expected or if we should fix

**Comment by luizstacio** [comment]
> I'm testing in latest PR preview docs/extension
> 
> context: I added a new network "Localhost" pointing
> 
> problems in connect screen:
> 
> * when click in "Localhost", nothing happens
> * when click in "account" nothing happends
> * after connecting, when trying to disconnect, an error is shown
> 
>  Screen.Recording.2023-01-12.at.16.14.38.mov

- The network selector was disabled, which is not supposed to work, we should remove the cursor pointer also, and will do it.
- The error about the network being blocked is on the browser level you can enable it by clicking on the Brave shield on the side of the URL.
- The accounts button is not related to this PR but I open an issue for it: https://github.com/FuelLabs/fuels-wallet/issues/432

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
code looks ok but I asked functional changes in prev comments... pls ask review again when finished @luizstacio

File: packages/sdk/src/Fuel.ts [hunk 1]
```diff
@@ -1,16 +1,40 @@
...
+      return FuelWeb3Privates.provider;
+    }
+    // Otherwise, create a new provider instance
+    // fetch the current network and connect the provider
+    const provider = new FuelWalletProvider(providerConfig.url, this);
+    FuelWeb3Privates.provider = provider;
+
+    // Listen for network changes and connect the provider
+    // selected network from the user
+    this.on('network', async (network) => {
```
Comment (matt-user) [inline packages/sdk/src/Fuel.ts:28]: Would it be possible for the network to change without a call to `getProvider`?  Basically I'm asking would it be possible for a network change event to fire before reaching this code which handles network change events.  Sorry if this is a dumb question lol.

Comment (luizstacio) [reply packages/sdk/src/Fuel.ts:28]: As the getProvider always returns the same instance when the event is triggered and the URL has changed, the changes will have an instant effect.

If the event is triggered before riching this point would not be a problem as the current network is requested from the wallet at the time we create the instance of the provider line 16.

**Review by matt-user — APPROVED** [review]
just tested locally and it works 😄

**Review by LuizAsFight — APPROVED** [review]
lgtm

**Pull request merged**